.phpcs.xml.dist

```diff
@@ -29,15 +16,14 @@
 		</properties>
 	</rule>
 
-	<!-- Check all PHP files in directory tree by default. -->
-	<arg name="extensions" value="php"/>
-	<file>.</file>
-
-	<!-- Show sniff codes in all reports -->
-	<arg value="s"/>
-
-	<arg name="colors"/>
-
-	<exclude-pattern>/node_modules/*</exclude-pattern>
-	<exclude-pattern>/vendor/*</exclude-pattern>
+	<!-- For the coding-standards package, we want to follow PHPCS convention, so excluding some WP rules. -->
```

Comment:
"PHPCS convention" isn't a _thing_.

If you're excluding rules here, what rules are you following, with respect to file naming, etc.

Reply:
To be honest, I wrote the code looking at other examples of custom sniffs (as this is my first so stumbling through the dark a bit), then ran it through `phpcs`. It threw a good number of alerts and I noticed that WPCS itself excludes a few rules, so I copied that.

https://github.com/WordPress/WordPress-Coding-Standards/blob/develop/.phpcs.xml.dist#L17-L20

Reply:
Trying this now. We need to exclude the filename rule because PHPCS seemingly requires that a sniff ends in `Sniff.php`, which would fail two WPCS rules:
```
 1 | ERROR | Filenames should be all lowercase with hyphens as word separators. Expected class-hooksmusthavedocblocksniff.php, but found class-hooksmusthavedocblockSniff.php. (WordPress.Files.FileName.NotHyphenatedLowercase)
 1 | ERROR | Class file names should be based on the class name with "class-" prepended. Expected class-hooksmusthavedocblocksniff.php, but found class-hooksmusthavedocblockSniff.php.
   |       | (WordPress.Files.FileName.InvalidClassFileName)
```

Also, need to exclude the var naming rule from WPCS because of `Tokens::$emptyTokens` and `$this->phpcsFile`.

The third exclusion doesn't make a difference right now, but added it since WPCS had it.

Will update the comment to make it clearer I'm borrowing from WPCS, not following "PHPCS convention".

Reply:
2a7128a5e

Reply:
Ah - the "PHPCS convention" bit makes sense now; really "the PHPCS sniff class file naming convention (`...Sniff.php`)" :-)